pandas/io/feather_format.py

```diff
@@ -26,7 +28,7 @@
 @doc(storage_options=generic._shared_docs["storage_options"])
 def to_feather(
     df: DataFrame,
-    path: FilePathOrBuffer[bytes],
+    path: FilePath | WriteBuffer[bytes],
```

Comment:
Are we going to have to document these new types?  Because the docs will show the argument as `FilePath | WriteBuffer[bytes]` in the signature, but the docs (in this case) show `path : string file path, or file-like object` so I'm worried about confusion when people see that the signature includes a type that isn't documented

Reply:
Good point! I think there are probably at least two options to avoid this confusion:

- create a new section in one of the IO overviews that contains definitions for these `*Buffer`s, then update the doc-string to point to the definition page
- Luckily, most to/read functions work with a `Read/WriteBuffer`: updating the doc-string to "str/byte readable/writeable file-like object" might be sufficient for them? For functions that have more specific needs (probably only read_csv; truncate (excel) and readlines (pickle) are also fairly standard?) update the doc-string to outline the "unusual" requirements (`__iter__` for the python engine).

I don't have strong opinions about this, except that the doc-strings need to be updated to some degree :)

Reply:
Maybe do both?  Create a section that explains the types (in case anyone searches for them), but make all the doc strings explicit, maybe something like "`ReadBuffer[bytes] (str/byte readable file-like object)"

Reply:
> Because the docs will show the argument as...

I'm not sure what is meant by this, the docs don't show the type-hints.

https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.DataFrame.to_feather.html

Reply:
The current docs don't because we don't have type hints in the 1.3.4, but I think they will show in the future with all the typing work that has been done.  But I could be wrong about that.

If they are not showing up, we should discuss whether they should be showing for the entire docs as we keep adding typing to the method/function signatures.

Reply:
Agreed that this is an issue that is becoming more and more prominent, and something that should be discussed. However, it seems to me that this is already pervasive throughout the code (e.g. "array-like") and this PR doesn't make the issue significantly worse.

Reply:
I updated the doc-strings to be consistent (without outlining the exact requirements).

Having an overview page of all typing definitions (not just IO) would be great (and probalby also necesary/expected when pandas becomes a py.typed library).

I know that google-conform doc-strings allow to skip the type descritpion if type annotatiosn are present. That would at least ensure that the doc-string and the type annotations are in sync.